File: src/app/shared/support/interfaces/transactions/transactionOutput.ts
import { TransactionType } from './../../enums/transactionTypes/transaction-types';
export interface TransactionOutput {
  id: string;
  description: string;
  date: Date;
  value: number;
  categoryId: string;
  receiverCustomerId: string;
  senderCustomerId?:string;
  transactionType: TransactionType;
}

File: backend/finan-exp-backend/Transactions/Dtos/TransactionOutput.cs
using Shared.Enums;
using Transactions.Models;

namespace Transactions.Dtos
{
  public class TransactionOutput : ITransactionModel
  {
    public Guid Id { get; set; }
    public string Description { get; set; }

    public DateTime Date { get; set; }

    public decimal Value { get; set; }

    public Guid? CategoryId { get; set; }

    public Guid ReceiverCustomerId { get; set; }

    public Guid? SenderCustomerId { get; set; }

    public TransactionType TransactionType { get; set; }
  }
}
